codec/src/test/java/io/netty/handler/codec/compression/ZlibTest.java

```diff
@@ -111,6 +113,68 @@ public void testGZIP2() throws Exception {
         }
     }
 
+    @Test
+    public void testStreamingGZip() throws Exception {
+        ByteBuf data = PooledByteBufAllocator.DEFAULT.heapBuffer(12);
+        ByteBuf data2 = PooledByteBufAllocator.DEFAULT.heapBuffer(13);
+        data.writeBytes("hello, world".getBytes(CharsetUtil.UTF_8));
+        data2.writeBytes("hello, heaven".getBytes(CharsetUtil.UTF_8));
+
+        EmbeddedChannel chEncoderGZip = new EmbeddedChannel(createEncoder(ZlibWrapper.GZIP));
+        EmbeddedChannel chDecoderGZip = new EmbeddedChannel(createDecoder(ZlibWrapper.GZIP, true));
+
+        try {
+            chEncoderGZip.writeOutbound(data.copy());
+            chDecoderGZip.writeInbound(chEncoderGZip.readOutbound());
+            ByteBuf buf = chDecoderGZip.readInbound();
+            assertEquals(buf, data);
+
+            chEncoderGZip.writeOutbound(data2.copy());
+            chDecoderGZip.writeInbound(chEncoderGZip.readOutbound());
+            ByteBuf buf2 = chDecoderGZip.readInbound();
+            assertEquals(buf2, data2);
+
+            data.release();
+            buf.release();
+            data2.release();
+            buf2.release();
+        } finally {
+            dispose(chDecoderGZip);
+            dispose(chEncoderGZip);
+        }
+    }
+
+    @Test
+    public void testStreamingZlib() throws Exception {
+        ByteBuf data = PooledByteBufAllocator.DEFAULT.heapBuffer(12);
+        ByteBuf data2 = PooledByteBufAllocator.DEFAULT.heapBuffer(13);
+        data.writeBytes("hello, world".getBytes(CharsetUtil.UTF_8));
+        data2.writeBytes("hello, heaven".getBytes(CharsetUtil.UTF_8));
+
+        EmbeddedChannel chEncoderGZip = new EmbeddedChannel(createEncoder(ZlibWrapper.ZLIB));
```

Comment:
Can you create a method to extract out the commonalities between this method and `testStreamingGZip`?  Seems like the only difference is this ENUM value.